src/Data/Nat/Divisibility/Core.agda

```diff
@@ -48,9 +49,14 @@ pattern divides-refl q = divides q refl
 -- Basic properties
 
 *-pres-∣ : ∀ {m n o p} → o ∣ m → p ∣ n → o * p ∣ m * n
-*-pres-∣ {m} {n} {o} {p} (divides c m≡c*o) (divides d n≡d*p) =
+*-pres-∣ {m@.(c * o)} {n@.(d * p)} {o} {p} (divides-refl c) (divides-refl d) =
   divides (c * d) (begin
-    m * n             ≡⟨ cong₂ _*_ m≡c*o n≡d*p ⟩
+    m * n             ≡⟨⟩
     (c * o) * (d * p) ≡⟨ [m*n]*[o*p]≡[m*o]*[n*p] c o d p ⟩
     (c * d) * (o * p) ∎)
   where open ≡-Reasoning
```

Comment:
The introduction of the pattern synonym makes this lemma largely trivial, and it has only a single usage in `Data.Nat.DivMod./-*-interchange`, where moreover the interchange employed in the proof of the lemma is then undone by that use, so it seems as though this could even be dispensed with altogether? (and moved to `Data.Nat.Divisibility`... with suitable deprecation?)